### PR TITLE
Prevent green from hanging on internal failure during error reporting.

### PR DESCRIPTION
**Pull request by charles-l**
`green` currently hangs when trying to run with a doctests in the load_tests call:

(save the following to a file named `test_doctest.py`):
```
import unittest                                                                                                                                                                                                                                                                                                                
import doctest                                                                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                                                               
def f():                                                                                                                                                                                                                                                                                                                       
    '''                                                                                                                                                                                                                                                                                                                        
    >>> f()                                                                                                                                                                                                                                                                                                                    
    2                                                                                                                                                                                                                                                                                                                          
    '''                                                                                                                                                                                                                                                                                                                        
    return 2                                                                                                                                                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                                                               
def load_tests(loader, tests, ignore):                                                                                                                                                                                                                                                                                         
    return doctest.DocTestSuite('test_doctest')
```

This patch catches the exception that's thrown in the worker and reports it back. Before, the workers would bail and become defunct processes, while the scheduler process would hang because it was waiting for a result from the queue (which never showed up)

This doesn't fix the underlying issue with doctest (unfortunately), but it at least prints the error and prevents `green` from hanging.

### CODE REVIEW

File: green/process.py [hunk 1]
```diff
@@ -327,11 +330,15 @@ def finalize_callback(test_result):
...
-                result.stopTest(test)
-                queue.put(result)
+                try:
+                    err = sys.exc_info()
+                    result.startTest(test)
+                    result.addError(test, err)
+                    result.stopTest(test)
+                    queue.put(result)
+                except:
+                    raise_internal_failure('Green encoundered an error when running the test.')
```
Comment (CleanCut) [inline green/process.py]:
```suggestion
                    raise_internal_failure('Green encountered an internal error when running the test.  This error is known to occur with doctests, which Green does not yet support.')
```

**Comment by charles-l** [comment]
Oh, interesting -- it looks like the bug is actually in the GreenTestLoader class. When I run `poolRunner('test_doctest', ...)` (so it imports it directly without wrapping in the GreenTestLoader), the tests run as expected. I'll try and dig into that issue and see if I can't fix it...

**Comment by CleanCut** [comment]
Fascinating!  Testing for the win... 😉  I'm excited to hear what you find.

**Comment by CleanCut** [comment]
Fun fact: I started looking into this and ended up _almost_ adding doctest support to green.  I've just got a final issue that I'm stuck on that is driving me crazy.

**Comment by CleanCut** [comment]
Got doctest support working!  I want to finish writing up some new tests for it (naturally), and then I'll put together a new release.

**Comment by CleanCut** [comment]
Support for doctests was added in version 3.1.0, just released.

**Comment by charles-l** [comment]
Oh sweet -- thanks for adding this support! I'm excited to start using it :)

**Comment by CleanCut** [comment]
Let me know how it goes!

**Pull request merged**